### PR TITLE
test(ContentTypeMiddleware): add content-type middleware

### LINKED ISSUE #192: content-type middleware
add a middleware which automatically adds content type (should accept content-type header from constructor, and default to `application/json`

### PR DESCRIPTION
**Pull request by cyberhck**
closes #192

### CODE REVIEW

**Comment by cyberhck** [comment]
cc @paibamboo 🙂

File: src/middlewares/ContentTypeMiddleware.spec.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {ContentTypeMiddleware} from "./ContentTypeMiddleware";
+import {IFetchRequest} from "./FetchMiddleware";
+
+describe("ContentTypeMiddleware", () => {
+    it("should add a content type application/json by default", () => {
+        const spy = jest.fn((options: IFetchRequest) => {
+            const contentType = options.headers && options.headers["content-type"];
+            expect(contentType).toBe("application/json");
```
Comment (paibamboo) [inline src/middlewares/ContentTypeMiddleware.spec.ts:8]: Is this `expect ` even gets called? Should we do [`expect.assertions`](https://jestjs.io/docs/en/asynchronous) to ensure that?

Comment (cyberhck) [reply src/middlewares/ContentTypeMiddleware.spec.ts:8]: I took that from another spec file, but let me verify that tonight :) (october just started, so...)

Comment (cyberhck) [reply src/middlewares/ContentTypeMiddleware.spec.ts:8]: image

it does fail if I remove the text/plain, so the expect does work, I took that as an example from another test,

**Pull request merged**